compiler/rustc_hir_typeck/src/intrinsicck.rs

```diff
@@ -83,6 +83,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let skeleton_string = |ty: Ty<'tcx>, sk| match sk {
             Ok(SizeSkeleton::Known(size)) => format!("{} bits", size.bits()),
             Ok(SizeSkeleton::Pointer { tail, .. }) => format!("pointer to `{tail}`"),
+            Ok(SizeSkeleton::Generic(size)) => {
+                if let Some(size) = size.try_eval_usize(tcx, self.param_env) {
+                    format!("{size} bytes")
+                } else {
+                    format!("generic size")
```

Comment:
Is there no way to actually print the expression? "generic size != generic size" feels like very bad UX.

Reply:
For now, trying to print out an expression just prints `[const expr]`, which could be fixed but not sure if it should be tacked onto this pr

Reply:
you should just print the consts so that when printing for `ConstKind::Expr` functions correctly this code will print correctly. please dont implement correct formatting for `ConstKind::Expr` in this PR